pkg/controller/authentication/authentication_controller.go

```diff
@@ -0,0 +1,269 @@
+	}
+	r.status.OnCRFound()
+	reqLogger.V(2).Info("Loaded config", "config", authentication)
+
+	// Query for the installation object.
+	network, err := installation.GetInstallation(context.Background(), r.client)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Error(err, "Installation not found")
+			r.status.SetDegraded("Installation not found", err.Error())
+			return reconcile.Result{}, err
+		}
+		log.Error(err, "Error querying installation")
+		r.status.SetDegraded("Error querying installation", err.Error())
+		return reconcile.Result{}, err
+	}
+	if network.Status.Variant != oprv1.TigeraSecureEnterprise {
+		log.Error(err, fmt.Sprintf("Waiting for network to be %s", oprv1.TigeraSecureEnterprise))
+		r.status.SetDegraded(fmt.Sprintf("Waiting for network to be %s", oprv1.TigeraSecureEnterprise), "")
+		return reconcile.Result{}, nil
+	}
+
+	// Make sure authentication and managementclusterconnection are not present at the same time.
+	_, err = utils.GetManagementClusterConnection(ctx, r.client)
+	if err == nil {
+		log.Error(err, "Having Authentication CR and ManagementCluster CR at the same time is not supported")
+		r.status.SetDegraded("Having Authentication CR and ManagementCluster CR at the same time is not supported", err.Error())
+		return reconcile.Result{}, err
+	} else if !errors.IsNotFound(err) {
+		log.Error(err, "Error querying ManagementCluster")
+		r.status.SetDegraded("Error querying ManagementCluster", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	// Cert used for TLS between voltron and dex when voltron is proxying dex from https://<manager-url>/dex
+	tlsSecret := &corev1.Secret{}
+	if err := r.client.Get(ctx, types.NamespacedName{Name: render.DexTLSSecretName, Namespace: render.OperatorNamespace()}, tlsSecret); err != nil {
+		if errors.IsNotFound(err) {
+			// We need to render a new one.
+			tlsSecret = render.CreateDexTLSSecret()
+		} else {
+			log.Error(err, "Failed to read dex tls secret")
+			r.status.SetDegraded("Failed to read dex tls secret", err.Error())
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Dex will be configured with the contents of this secret, such as clientID and clientSecret.
+	idpSecret, err := getIdpSecret(ctx, r.client, authentication)
+	if err != nil {
+		log.Error(err, "Invalid or missing idp secret")
+		r.status.SetDegraded("Invalid or missing idp secret", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	// Set namespace for secrets so they can be used in the namespace of dex.
+	idpSecret = render.CopySecrets(render.DexNamespace, idpSecret)[0]
+
+	dexSecret := &corev1.Secret{}
+	if err := r.client.Get(ctx, types.NamespacedName{Name: render.DexObjectName, Namespace: render.OperatorNamespace()}, dexSecret); err != nil {
+		if errors.IsNotFound(err) {
+			// We need to render a new one.
+			dexSecret = render.CreateDexClientSecret()
+		} else {
+			log.Error(err, "Failed to read dex secret")
+			r.status.SetDegraded("Failed to read dex secret", err.Error())
```

Comment:
"failed to read secret tigera-operator/tigera-dex" will be more useful - spells out exactly the namespace / name of the object that caused the error.